File: leetcode/python/plus one/plus_one.py
class Solution:
    # @param digits, a list of integer digits
    # @return a list of integer digits
    def plusOne(self, digits):
        digits.reverse()
        res = []
        carry = (digits[0] + 1) / 10
        t = (digits[0] + 1) % 10
        res.append(t)
        for i in digits[1:]:
            t = (i + carry ) % 10
            carry = (i + carry ) / 10
            res.append(t)
        if carry == 1:
            res.append(1)
        res.reverse()
        return res
            

solu = Solution()
solu.plusOne([0])

File: leetcode/cpp/PlusOne.cpp
//
// Created by zz on 2017/10/8.
// Time : O(n2)
// Space : O(1)

#include <vector>

using namespace std;

class Solution {
public:
    vector<int> plusOne(vector<int> &digits) {
        vector<int> ans(digits);
        int carry = 1;
        for (auto iter = ans.rbegin(); iter != ans.rend(); iter++) {
            *iter += carry;
            carry = *iter / 10;
            *iter = *iter % 10;
        }
        if (carry != 0) {
            ans.emplace(ans.begin(), carry);
        }
        return ans;

    }
};